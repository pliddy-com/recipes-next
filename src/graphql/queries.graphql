# Query for Homepage RecipeCollection (all recipes)

query homePageQuery {
  recipeCollection(order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe collection based on filter

query recipeCollection($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    total
    items {
      ...RecipeDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)

query listPageQuery($where: TagFilter) {
  tagCollection(where: $where, limit: 1) {
    total
    items {
      slug
      title
      linkedFrom {
        recipeCollection(limit: 24) {
          total
          items {
            ...RecipeSummary
          }
        }
      }
    }
  }
}

# Query for recipe slugs

query recipeSlugsCollection($where: RecipeFilter) {
  recipeCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query categorySlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query tagSlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
      linkedFrom {
        recipeCollection {
          total
        }
      }
    }
  }
}

# query for Taxonomy with Category & Subcategory levels

query taxonomyCollection($where: TaxonomyFilter) {
  taxonomyCollection(where: $where, limit: 1) {
    items {
      ...TaxonomyDefault

      tag {
        ...TagDefault
      }

      childrenCollection(limit: 24) {
        total
        items {
          ... on Tag {
            ...TagDefault

            linkedFrom {
              recipeCollection(limit: 1) {
                total
              }
            }
          }
          ...TaxonomyWithChildren
        }
      }
    }
  }
}
