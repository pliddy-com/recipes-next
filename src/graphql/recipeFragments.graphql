# Fragments with default fields for Recipe-specific sections:
#   * Ingredients
#   * Instructions
#   * Image
#   * Recipe

fragment IngredientsDefault on IngredientSection {
  sys {
    id
  }
  title
  slug
  label
  ingredientList
}

fragment InstructionsDefault on InstructionSection {
  sys {
    id
  }
  title
  slug
  label
  instructionList
}

fragment ImageDefault on Asset {
  sys {
    id
  }
  __typename
  title
  description
  contentType
  fileName
  size
  url
  height
  width
}

fragment DescriptionDefault on Recipe {
  description {
    json
    links {
      entries {
        inline {
          __typename
          sys {
            id
            __typename
          }
        }
        hyperlink {
          __typename
          sys {
            id
            __typename
          }
        }
        block {
          __typename

          sys {
            id
            __typename
          }
        }
      }
      assets {
        hyperlink {
          __typename
          title
        }
        block {
          __typename
          title
        }
      }
    }
  }
}

fragment RecipeDefault on Recipe {
  sys {
    id
  }
  __typename
  title
  slug
  description {
    json
  }
  abstract
  image {
    ...ImageDefault
  }
  ingredientsCollection(limit: 10) {
    items {
      ...IngredientsDefault
    }
  }
  equipment
  instructionsCollection(limit: 10) {
    items {
      ...InstructionsDefault
    }
  }
  notes
  tagsCollection(limit: 10) {
    items {
      ...TagDefault
    }
  }
}
