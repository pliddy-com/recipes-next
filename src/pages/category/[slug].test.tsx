// import testing-library methods
import { render } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import CategoryPage from './[slug]';

import { ListPageItemFragment } from 'types/generated/graphql';

// import config object to mock
const config = jest.requireMock('lib/config');

// set up default mock config object
jest.mock('lib/config', () => ({
  microcopy: {
    index: {
      defaultTitle: 'Default Index Title',
      description: 'Index page description',
    },
    site: {
      title: 'Site Title',
    },
  },
}));

// import api library to mock
// const api = jest.requireMock('lib/api');

// set up default mock api object
// jest.mock('lib/api', () => ({
//   queryListPageContent: jest
//     .fn()
//     .mockResolvedValue({ data: 'list page content' }),
//   queryCategorySlugs: jest.fn().mockResolvedValue({ data: 'category slugs' }),
// }));

describe('CategoryPage in category/[slug].tsx', () => {
  describe('when there is page content', () => {
    it('it renders the CategoryPage if there is content', () => {
      const pageContent = {
        slug: 'category-title',
        title: 'Category Title',
        linkedFrom: {
          recipeCollection: {
            total: 1,
            items: [
              {
                sys: {
                  id: 'sysid-0',
                  __typename: 'Sys',
                },
                __typename: 'Recipe',
                title: 'Recipe 1 Title',
                slug: 'recipe-1-title',
                abstract: 'Recipe 1 abstract.',
                image: {
                  sys: {
                    id: 'sysid-1',
                    __typename: 'Sys',
                  },
                  __typename: 'Asset',
                  title: 'Image 1 Title',
                  description: 'Image 1 description',
                  contentType: 'image/jpeg',
                  fileName: 'image1.jpg',
                  size: 99999,
                  url: 'https://test.url/biscuits.jpg',
                  height: 300,
                  width: 400,
                },
                tagsCollection: {
                  items: [
                    {
                      sys: {
                        id: 'sysid-2',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Tag 1',
                      slug: 'tag-1',
                    },
                    {
                      sys: {
                        id: 'sysid-3',
                        __typename: 'Sys',
                      },
                      __typename: 'Tag',
                      title: 'Tag 2',
                      slug: 'tag-2',
                    },
                  ],
                  __typename: 'RecipeTagsCollection',
                },
              },
            ],
            __typename: 'RecipeCollection',
          },
          __typename: 'TagLinkingCollections',
        },
        __typename: 'Tag',
      };

      const { container } = render(
        <CategoryPage
          pageContent={pageContent as ListPageItemFragment}
          preview={false}
        />
      );

      // assert that page head tags are rendered
      const titleTag = document.getElementsByTagName('title')[0];
      expect(titleTag).toBeInTheDocument();
      expect(titleTag.text.includes(pageContent.title));

      // assert that page container is rendered
      const page = document.querySelector('.page');
      expect(page).toBeInTheDocument();

      // assert that the component matches the existing snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe('when there is no content', () => {
    // before each test, delete microcopy node from config
    beforeEach(() => {
      delete config.microcopy;
    });

    it('it does not render the page', () => {
      const pageContent = undefined as unknown as ListPageItemFragment;

      render(<CategoryPage pageContent={pageContent} preview={false} />);

      // test if card compoent is not rendered
      const page = document.querySelector('.page');
      expect(page).toBeNull();
    });
  });

  describe('when there is no config object', () => {
    it('it does not render the page', () => {
      render(
        <CategoryPage
          preview={false}
          pageContent={{
            __typename: undefined,
            slug: undefined,
            title: undefined,
            linkedFrom: undefined,
          }}
        />
      );

      // assert that page container is not rendered
      const page = document.querySelector('.page');
      expect(page).toBeNull();
    });
  });
});
