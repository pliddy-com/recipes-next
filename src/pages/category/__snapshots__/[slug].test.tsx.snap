// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage in category/[slug].tsx when there is page content it renders the CategoryPage if there is content 1`] = `
<div>
  <title>
    Site Title | Category Title
  </title>
  <meta
    content="undefined Category Title"
    name="description"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg page css-1oqqzyl-MuiContainer-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Category Title
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-1wniyei-MuiTypography-root"
    >
      1 Total
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 css-wq3jbm-MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
